src/AWS/Lambda/Runtime.hs

```diff
@@ -80,6 +88,7 @@ import           Data.Aeson               (FromJSON, ToJSON)
 --     main :: IO ()
 --     main = runReaderTLambdaContext (evalStateT (mRuntimeWithContext myHandler) 0)
 -- @
+{-# DEPRECATED mRuntimeWithContext "mRuntimeWithContext will be replaced with mRuntimeWithContext' in future versions.  The original signature was problematic in that it pretended the LambdaContext was always available, when that is not the case." #-}
```

Comment:
My reading of the deprecation warning makes me think it's unclear which way the replacement is going. Perhaps clarify that this `mRuntimeWithContext` will be removed, and `mRuntimeWithContext'` will be renamed?

Reply:
Good call out--it's a bit awkward given that the names are so close.  Any suggestions on the wording?  I'm also happy to pull in commits if you fork this branch :)

Reply:
"mRuntimeWithContext will be removed soon, and mRuntimeWithContext' renamed to take its place. The original mRuntimeWithContext was problematic - see the haddocks for more details"

And then move the explanation of `LambdaContext` into the haddocks of the to-be-removed `mRuntimeWithContext`.